src/ObjectUtils/PdbFileDia.cpp

```diff
@@ -81,44 +82,53 @@ ErrorMessageOr<void> PdbFileDia::LoadDataForPDB() {
   return outcome::success();
 }
 
-ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> PdbFileDia::LoadDebugSymbols() {
+ErrorMessageOr<DebugSymbols> PdbFileDia::LoadRawDebugSymbols() {
+  ORBIT_SCOPE_FUNCTION;
+  DebugSymbols debug_symbols;
+  debug_symbols.load_bias = object_file_info_.load_bias;
+  debug_symbols.symbols_file_path = file_path_.string();
+
+  OUTCOME_TRY(LoadProcSymbols(SymTagFunction, debug_symbols));
+
+  return debug_symbols;
+}
+
+ErrorMessageOr<void> PdbFileDia::LoadProcSymbols(const enum SymTagEnum symbol_tag,
+                                                 DebugSymbols& debug_symbols) {
   HRESULT result = 0;
   CComPtr<IDiaEnumSymbols> dia_enum_symbols;
-  result = dia_global_scope_symbol_->findChildren(SymTagFunction, NULL, nsNone, &dia_enum_symbols);
+  result = dia_global_scope_symbol_->findChildren(symbol_tag, /*name=*/nullptr,
+                                                  NameSearchOptions::nsNone, &dia_enum_symbols);
   if (result != S_OK) {
     return ErrorMessage{
         absl::StrFormat("findChildren failed for %s (%u)", file_path_.string(), result)};
   }
 
-  ModuleSymbols module_symbols;
-  module_symbols.set_load_bias(object_file_info_.load_bias);
-  module_symbols.set_symbols_file_path(file_path_.string());
   IDiaSymbol* dia_symbol = nullptr;
   ULONG celt = 0;
 
   while (SUCCEEDED(dia_enum_symbols->Next(1, &dia_symbol, &celt)) && (celt == 1)) {
     CComPtr<IDiaSymbol> dia_symbol_com_ptr = dia_symbol;
-    SymbolInfo symbol_info;
 
     BSTR function_name = {};
     if (dia_symbol->get_name(&function_name) != S_OK) continue;
     std::wstring name(function_name);
-    symbol_info.set_name(std::string(name.begin(), name.end()));
-    symbol_info.set_demangled_name(llvm::demangle(symbol_info.name()));
     SysFreeString(function_name);
 
+    FunctionSymbol& function_symbol = debug_symbols.function_symbols.emplace_back();
+    function_symbol.demangled_name = std::string(name.begin(), name.end());
```

Comment:
You are not setting the mangled name here. I am wondering if this leads to confusion issues later on?

Reply:
Same for the llvm implementation.